### PR TITLE
Prototype of Mutex-protected crypt module

### PR DESCRIPTION
**Pull request by jbaublitz**
Related to #2277 
Related to #2382 
Related to #2355

### CODE REVIEW

**Comment by jbaublitz** [comment]
CI passed! Putting up for review.

File: src/engine/strat_engine/backstore/crypt.rs [hunk 1]
```diff
@@ -330,16 +358,18 @@ impl CryptActivationHandle {
...
 
-        can_unlock_with_failures(physical_path)
-            .map_err(|e| {
+        lock.and_then(|_| {
+            can_unlock_with_failures(physical_path).map_err(|e| {
                 warn!(
                     "stratisd was unable to simulate opening the given device \
-                    in the current environment. This may be due to an expired \
-                    or missing key in the kernel keyring: {}",
+                        in the current environment. This may be due to an expired \
```
Comment (mulkieran) [inline src/engine/strat_engine/backstore/crypt.rs:365]: Wierd looking extra spaces.

**Review by mulkieran — CHANGES_REQUESTED** [review]
This looks correct to me, and I believe the mutex approach will be less restrictive than the assertion approach but just as correct. There is one unnecessary formatting change, I think.

File: src/engine/strat_engine/backstore/crypt.rs (hunk 1 above)
Comment (jbaublitz) [reply src/engine/strat_engine/backstore/crypt.rs]: Resolved.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Looks ready then. Please close this PR and open a new one for final review.

**Comment by jbaublitz** [comment]
I need to make one last change to this. I'll explain in stand up.

**Comment by jbaublitz** [comment]
Obsoleted by stratis-storage/libcryptsetup-rs#121

**Pull request closed without merging**